### PR TITLE
111

### PR DESCRIPTION
**Pull request by melnikalex1977**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,36 @@
-# write your code here
+def calculate_team_total_rating(team: list) -> int:
+    calculate_total_rating = 0
+    for yy in team:
```
Comment (vasylhnatiuk) [inline app/main.py:3]: yy?? what is it? rename it here and all of the code

File: app/main.py [hunk 2]
```diff
@@ -1 +1,36 @@
-# write your code here
+def calculate_team_total_rating(team: list) -> int:
```
Comment (vasylhnatiuk) [inline app/main.py:1]: In general, this function code is so bad. Why don't you use get_rating() to get a rating? Rewrite it and use sum() and get_rating().

File: app/main.py [hunk 3]
```diff
@@ -1 +1,36 @@
...
+    return calculate_total_rating
+
+
+def elves_concert(elves: list) -> int:
+    for yy in elves:
+        print(f"{yy.nickname} is playing a song on the "
+              f"{yy.musical_instrument}")
+
+
+def play_elf_song(elves: list) -> int:
```
Comment (vasylhnatiuk) [inline app/main.py:23]: why do you need this?

File: app/main.py [hunk 4]
```diff
@@ -1 +1,36 @@
...
+        print(f"{yy.nickname} is playing a song on the "
+              f"{yy.musical_instrument}")
+
+
+def feast_of_the_dwarves(dwarves: list) -> int:
+    for yy in dwarves:
+        print(f"{yy.nickname} is eating {yy.favourite_dish}")
+
+
+def eat_favourite_dish(dwarves: list) -> int:
```
Comment (vasylhnatiuk) [inline app/main.py:34]: why do you need this?

File: app/main.py [hunk 5]
```diff
@@ -1 +1,36 @@
...
+              f"{yy.musical_instrument}")
+
+
+def play_elf_song(elves: list) -> int:
+    for yy in elves:
+        print(f"{yy.nickname} is playing a song on the "
+              f"{yy.musical_instrument}")
+
+
+def feast_of_the_dwarves(dwarves: list) -> int:
```
Comment (vasylhnatiuk) [inline app/main.py:29]:
```suggestion
def feast_of_the_dwarves(dwarves: list[Dwarf]) -> int:
```

File: app/main.py [hunk 6]
```diff
@@ -1 +1,36 @@
...
+        elif type(yy).__name__ == "DwarfWarrior":
+            calculate_total_rating += yy.hummer_level + 4
+        elif type(yy).__name__ == "DwarfBlacksmith":
+            calculate_total_rating += yy.skill_level
+        elif yy == []:
+            calculate_total_rating = 0
+    return calculate_total_rating
+
+
+def elves_concert(elves: list) -> int:
```
Comment (vasylhnatiuk) [inline app/main.py:17]:
```suggestion
def elves_concert(elves: list[Elf]) -> int:
```

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]
name of branch 111??? What? Fix it

also correct everything that the previous mentor wrote to you

**Pull request closed without merging**